src/components/drop-down/drop-down.tsx

```diff
@@ -70,16 +60,77 @@ export class DropDown extends React.Component<DropDownProps, {}> {
             'className': 'drop-down'
         });
 
+        const ToggleIcon = this.props.toggleIcon!;
+
         return (
-            <div data-automation-id="DROP_DOWN" {...rootProps}>
-                <DropDownInput selectedItem={this.props.selectedItem} onClick={this.props.onInputClick} />
-                <DropDownList
-                    selectedItem={this.props.selectedItem}
-                    items={this.props.items}
-                    open={!!this.props.open}
-                    onChange={this.onItemClick}
-                />
+            <div
+                {...rootProps}
+                onKeyDown={this.onKeyDown}
+                onFocus={this.onFocus}
+                tabIndex={this.props.tabIndex}
+                ref={dropdown => this.setState({dropdown})}
+            >
+                <div data-automation-id="DROP_DOWN_INPUT" onClick={this.toggleDropdown} className="drop-down-input">
+                    <span className="label">{this.props.value!}</span>
+                    <div className="caret" data-automation-id="ICON">
+                        <ToggleIcon />
+                    </div>
+                </div>
+                <Popup open={this.state.open && !this.props.disabled} anchor={this.state.dropdown}>
+                    <div className="root">
+                        <SelectionList
+                            data-automation-id="DROP_DOWN_LIST"
+                            className="drop-down-list"
+                            value={this.props.value}
+                            onChange={this.onItemClick!}
+                            dataSource={this.props.dataSource}
+                        >
+                            {this.props.children}
+                        </SelectionList>
+                    </div>
+                </Popup>
             </div>
         );
     }
+
+    private onFocus = () => {
+        if (this.props.openOnFocus) {
+            this.openDropdown();
+        }
+    }
+
+    private toggleDropdown = () => {
+        if (!this.props.disabled) {
+            this.setState({open: !this.state.open});
+        }
+    }
+
+    private openDropdown() {
+        if (!this.props.disabled) {
+            this.setState({open: true});
+        }
+    }
+
+    private closeDropdown() {
+        if (!this.props.disabled) {
+            this.setState({open: false});
+        }
+    }
+
+    private onKeyDown = (e: any) => {
```

Comment:
why 'any'?

Reply:
Not sure, I'll change it to be more specific